go/arrow/csv/reader.go

```diff
@@ -681,6 +691,36 @@ func (r *Reader) parseTime32(field array.Builder, str string, unit arrow.TimeUni
 	field.(*array.Time32Builder).Append(val)
 }
 
+func (r *Reader) parseDecimal128(field array.Builder, str string, prec, scale int32) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+
+	val, err := decimal128.FromString(str, prec, scale)
+	if err != nil && r.err == nil {
+		r.err = err
+		field.AppendNull()
+		return
+	}
+	field.(*array.Decimal128Builder).Append(val)
+}
+
+func (r *Reader) parseDecimal256(field array.Builder, str string, prec, scale int32) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+
+	val, err := decimal256.FromString(str, prec, scale)
+	if err != nil && r.err == nil {
+		r.err = err
+		field.AppendNull()
+		return
+	}
+	field.(*array.Decimal256Builder).Append(val)
```

Comment:
the reason for the `r.err == nil` check is so that we don't lose the first error we find. The assumption being that if we get an error we're going to stop reading after this record and the data is bad anyways.

Reply:
if `err != nil` then `val` will be the default value, which is 0